python/cudf/cudf/_libxx/strings/translate.pyx

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2018-2020, NVIDIA CORPORATION.
+
+from libcpp.memory cimport unique_ptr
+from cudf._libxx.move cimport move
+
+from cudf._libxx.cpp.column.column cimport column
+from cudf._libxx.cpp.column.column_view cimport column_view
+from cudf._libxx.cpp.strings.translate cimport (
+    translate as cpp_translate
+)
+from cudf._libxx.column cimport Column
+from collections import OrderedDict
+from libcpp.vector cimport vector
+from libcpp.pair cimport pair
+from cudf._libxx.cpp.types cimport char_utf8
+
+
+def translate(Column source_strings,
+              object mapping_table):
+    """
+    Translates individual characters within each string
+    mapping present in the mapping_table.
+    """
+    cdef unique_ptr[column] c_result
+    cdef column_view source_view = source_strings.view()
+
+    mapping_table_dict = OrderedDict()
+
+    for key in mapping_table:
+        value = mapping_table[key]
+        if type(value) is str:
+            value = ord(value)
+        mapping_table_dict[(key, value)] = None
+
+    cdef vector[pair[char_utf8, char_utf8]] c_mapping_table
+    c_mapping_table = list(mapping_table_dict.keys())
```

Comment:
Cython handles translating a tuple into a pair without issue?

Reply:
Yep, it does.